code/scripting/api/objs/subsystem.cpp

```diff
@@ -23,6 +27,24 @@ ship_subsys_h::ship_subsys_h(object* objp_in, ship_subsys* sub) : object_h(objp_
 }
 bool ship_subsys_h::isSubsystemValid() const { return object_h::IsValid() && objp->type == OBJ_SHIP && ss != nullptr; }
 
+void ship_subsys_h::serialize(lua_State* /*L*/, const scripting::ade_table_entry& /*tableEntry*/, const luacpp::LuaValue& value, ubyte* data, int& packet_size) {
+	ship_subsys_h subsys;
+	value.getValue(l_Subsystem.Get(&subsys));
+	const ushort& netsig = subsys.IsValid() ? subsys.objp->net_signature : 0;
+	const int& subsys_index = subsys.isSubsystemValid() ? subsys.ss->parent_subsys_index : -1;
+	ADD_USHORT(netsig);
+	ADD_INT(subsys_index);
+}
+
+void ship_subsys_h::deserialize(lua_State* /*L*/, const scripting::ade_table_entry& /*tableEntry*/, char* data_ptr, ubyte* data, int& offset) {
+	ushort net_signature;
+	int subsys;
+	GET_USHORT(net_signature);
+	GET_INT(subsys);
+	object* obj = multi_get_network_object(net_signature);
+	new(data_ptr) ship_subsys_h(obj, obj == nullptr || subsys == -1 ? nullptr : ship_get_indexed_subsys(&Ships[obj->instance], subsys));
```

Comment:
Maybe I'm being paranoid, but maybe we should also check that the object type is in fact a ship, too.